resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -95,6 +108,23 @@ void query() {
         sendQuery(query);
     }
 
+    static void addRecord(DatagramDnsQuery query, DnsSection section, Object records) {
```

Comment:
Why go with this type unsafe way that forces you to type check and cast objects whose type you initially knew?